uni/lib/view/home/widgets/restaurant_card.dart

```diff
@@ -22,46 +39,55 @@ class RestaurantCard extends GenericCard {
 
   @override
   Widget buildCardContent(BuildContext context) {
-    return Consumer<RestaurantProvider>(
-        builder: (context, restaurantProvider, _) =>
-            RequestDependentWidgetBuilder(
-                context: context,
-                status: restaurantProvider.status,
-                contentGenerator: generateRestaurant,
-                content: restaurantProvider.restaurants,
-                contentChecker: restaurantProvider.restaurants.isNotEmpty,
-                onNullContent: Center(
-                    child: Text('Não existem cantinas para apresentar',
-                        style: Theme.of(context).textTheme.headline4,
-                        textAlign: TextAlign.center))));
+    return Consumer<RestaurantProvider>(builder: (context, restaurantProvider, _) {
+      final List<Restaurant> favoriteRestaurants = restaurantProvider.restaurants.where((restaurant) => restaurantProvider.favoriteRestaurants.contains(restaurant.name)).toList();
+      return RequestDependentWidgetBuilder(
+          context: context,
+          status: restaurantProvider.status,
+          contentGenerator: generateRestaurant,
+          content: favoriteRestaurants,
+          contentChecker: favoriteRestaurants.isNotEmpty,
+          onNullContent: Center(
+              child: Text('Não existem cantinas para apresentar',
```

Comment:
Suggestion: The phrase could be "Não existem restaurantes favoritos" and a button "Adicionar" appearts to redirect to "restaurantes" pages.

Like this [example](https://github.com/NIAEFEUP/project-schrodinger/blob/cf86d9273fb5d8044a229e43046802b6c2ab36dc/uni/lib/view/common_widgets/request_dependent_widget_builder.dart#L86-L97).

Reply:
![Screenshot 2023-03-16 at 14 27 38](https://user-images.githubusercontent.com/81827192/225648625-4a2dd76f-53a5-4277-be79-af532cfb9235.png)